core/sql/optimizer/PartFunc.cpp

```diff
@@ -5009,6 +5020,118 @@ void RangePartitioningFunction::print(FILE* ofd, const char* indent,
   partitionBoundaries_->print(ofd, indent, title);
 
 } // RangePartitioningFunction::print()
+
+
+NABoolean 
+compareEncodedKey(const char* low, const char* key, const char* high, Int32 keyLen, NABoolean checkLast)
```

Comment:
Yes it is true for all intervals except the last one, which is an
all-closed interval.

On Mon, Oct 19, 2015 at 1:18 PM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/optimizer/PartFunc.cpp
> <https://github.com/apache/incubator-trafodion/pull/132#discussion_r42406620>
> :
>
> > @@ -5009,6 +5020,118 @@ void RangePartitioningFunction::print(FILE* ofd, const char* indent,
> >    partitionBoundaries_->print(ofd, indent, title);
> >
> >  } // RangePartitioningFunction::print()
> > +
> > +
> > +NABoolean
> > +compareEncodedKey(const char* low, const char* key, const char* high, Int32 keyLen, NABoolean checkLast)
>
> So, this routine checks to see if the key is in the half-open interval
> [low,hi), correct?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/132/files#r42406620>.
>



-- 
Regards, --Qifan